src/Factors/GenericFunctions.jl

```diff
@@ -119,7 +124,8 @@ function getSample(cf::CalcFactor{<:ManifoldPrior}, N::Int=1)
     
     Xc = [rand(Z) for _ in 1:N]
     
-    X = get_vector.(Ref(M), Ref(p), Xc, Ref(DefaultOrthogonalBasis()))
+    # X = get_vector.(Ref(M), Ref(p), Xc, Ref(DefaultOrthogonalBasis()))
+    X = hat.(Ref(M), Ref(p), Xc)
     points = exp.(Ref(M), Ref(p), X)
```

Comment:
I would avoid using `Ref()` here during development.  I'd recommend we only add that after things are working.  I'm not sure if Ref will work here, especially with the multithreading code.  Don't believe it is necessary to do that optimization now.

Reply:
It's not an optimization, only for the broadcasting. I can use other containers such as `Tuple`, but prefer `Ref`.  I just find broadcasted code easier to follow and maintain than for loops.